### PR TITLE
Studying for degree

### PR DESCRIPTION
**Pull request by goodviber**
### JIRA ticket number
https://dfedigital.atlassian.net/browse/GITPB-122
### Changes proposed in this pull request
Studying for a degree and equivalent pathway

### CODE REVIEW

File: app/models/start_teacher_training.rb [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
   attribute :year_of_entry, :string
 
-  validate :date_cannot_be_in_the_past
+  validate :date_cannot_be_in_the_past, unless: :dont_know
   
   def year_range(number_of_years) # sets year range for view
-    (Date.today.year..Date.today.next_year(number_of_years).year).map do |year|
-      OpenStruct.new(value: year, name: year)
+    years = []
+    (Date.today.year..Date.today.next_year(number_of_years).year).each do |year|
```
Comment (ethax-ross) [inline app/models/start_teacher_training.rb:8]:
```
years = (Date.today.year..Date.today.next_year(number_of_years).year).map do |year|
  OpenStruct.new(value: year, name: year)
end
years << OpenStruct.new(value: 'dont know', name: "Don't know")
```

File: app/models/studying/primary_maths_english_grade4.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Studying
+  class PrimaryMathsEnglishGrade4 < PrimaryMathsEnglishGrade4
+    def next_step
+      if has_required_subjects == "yes"
```
Comment (ethax-ross) [inline app/models/studying/primary_maths_english_grade4.rb:4]: It looks like we use true/false in some of these and yes/no in others - can they be made consistent?

Comment (goodviber) [reply app/models/studying/primary_maths_english_grade4.rb]: I am converting all to true/false, as I encounter them. Have done this one.

File: app/models/start_teacher_training.rb (hunk 1 above)
Comment (goodviber) [reply app/models/start_teacher_training.rb]: 👍

**Pull request merged**